run_public_tests.py

```
@@ -0,0 +1,195 @@
+import os
+import subprocess
+import sys
+import traceback
+from datetime import datetime, timezone
+from pathlib import Path
+
+
+class PublicTestRunner:
+    """Manages and runs public-web E2E tests"""
+
+    def __init__(self):
```

Ruff (0.14.7)

Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)